### DISCUSSION TITLE
Load SPA from cache without network connection

### DISCUSSION THREAD

**Discussion by Urbanus01** [cefsharp/CefSharp#3332] [General]
Hello All,

**- What version of the product are you using?**

v86.0.24

**- What architecture x86 or x64?**

Both

**- On what operating system?**

Win10

**- Are you using WinForms, WPF or OffScreen?**

WPF

**- What steps will reproduce the problem?**

I faced with next issues:

SefSharp doesn't load SPA website from cache after close and open window (WPF, Cef version 86.0.24).
I try to open sites with old HTML manifest mechanizm or jakecache.js, 
this issue is reproducible in both cases.
I have already set up CachePath folder, and see cache files stored on my hard disk, but looks lake that they doesn't used, if no internet connection.

 public partial class MainWindow : Window
    {
        private readonly string CacheFolderPath = $"{Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)}\\POS Software\\Shell";

        public MainWindow()
        {
            CefSettings Settings = new CefSettings
            {
                CachePath = Path.Combine(CacheFolderPath, "CefSharp\\Cache")                

            };
            Settings.CefCommandLineArgs.Add("ignore-certificate-errors");

            if (Cef.IsInitialized == false)
            {
                Cef.Initialize(Settings);
            }
            CefSharpSettings.ConcurrentTaskExecution = true;
            InitializeComponent();
            var urltoLoad = DataInteractionService.GetValue(DataInteractionConstants.UrlToLoad);
          
            if (!string.IsNullOrWhiteSpace(urltoLoad))
            {
                this.GeneralChromiumWebBrowser.Address = urltoLoad;
            }
            this.GeneralChromiumWebBrowser.BrowserSettings = new BrowserSettings
            {
                LocalStorage = CefState.Enabled, WebSecurity = CefState.Disabled
               
            };
}

Could you assist, how to solve it?

**Comment by amaitland** [comment]
Does it work in Chrome when the internet is disconnected? If yes then you can try setting the browser to offline via DevTools see https://bitbucket.org/chromiumembedded/cef/issues/245/direct-control-over-online-offline-status for some background. 

You can try http://cefsharp.github.io/api/86.0.x/html/M_CefSharp_DevTools_Network_NetworkClient_EmulateNetworkConditionsAsync.htm

You can try adapt the example below
```c#
using (var client = chromiumWebBrowser.GetDevToolsClient())
{
    _ = client.Network.SetUserAgentOverrideAsync("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36 - Testing 123");
}
```

**Reply by Urbanus01** [reply]
Hello amaitland,

First of all, Thank you for your time!

>>Does it work in Chrome when the internet is disconnected?
Yes, it does. Please try to open this https://pos.ehopper.com/25 

I have tried all tricks that you mentioned, but unfortunately, it doesn't help =( Lent me more concretize the steps for prevent misunderstanding.

1) Open this link https://pos.ehopper.com/25 on Chrome or Cef WPF application( I set CachePath for CefSettings and apply it)
2) On Dev Tool in Network tab set Offline.
3) Refresh page. For Chrome ( Cnt + R on dev tool or just enter on address line). 
For Cef WPF aplication close and open (sets same link on your code) or (I use binded hot keys commanad for 
 this.GeneralChromiumWebBrowser.Address = url;           
 this.GeneralChromiumWebBrowser.Load(url);
).
4) In Chrome everything works fine, website loaded. For Cef application I see white screen and one line with ERR_INTERNET_DISCONNECTED on Network dev tool tab.

Also I try this with HTML5 cache manifest with same results. 

Best regards,
Mykhailo Dashkel

**Reply by amaitland** [reply]
You can try asking in https://magpcss.org/ceforum/viewforum.php?f=18 (CefSharp is just one of many chromium embedded framework wrappers, there's a much larger community on ceforum).

I'm now on holidays.
